x/community/types/proposal.go

```diff
@@ -192,3 +195,62 @@ func (p *CommunityCDPRepayDebtProposal) ValidateBasic() error {
 	}
 	return nil
 }
+
+// NewCommunityCDPRepayDebtProposal creates a new community pool cdp debt repay proposal.
+func NewCommunityCDPWithdrawCollateralProposal(
+	title string,
+	description string,
+	collateralType string,
+	collateral sdk.Coin,
+) *CommunityCDPWithdrawCollateralProposal {
+	return &CommunityCDPWithdrawCollateralProposal{
+		Title:          title,
+		Description:    description,
+		CollateralType: collateralType,
+		Collateral:     collateral,
+	}
+}
+
+// GetTitle returns the title of the proposal.
+func (p *CommunityCDPWithdrawCollateralProposal) GetTitle() string { return p.Title }
+
+// GetDescription returns the description of the proposal.
+func (p *CommunityCDPWithdrawCollateralProposal) GetDescription() string { return p.Description }
+
+// GetDescription returns the routing key of the proposal.
+func (p *CommunityCDPWithdrawCollateralProposal) ProposalRoute() string { return ModuleName }
+
+// ProposalType returns the type of the proposal.
+func (p *CommunityCDPWithdrawCollateralProposal) ProposalType() string {
+	return ProposalTypeCommunityPoolCDPWithdrawCollateral
+}
+
+// String implements fmt.Stringer
+func (p *CommunityCDPWithdrawCollateralProposal) String() string {
+	var b strings.Builder
+	b.WriteString(fmt.Sprintf(`Community CDP Withdraw Collateral Proposal:
+  Title:           %s
+  Description:     %s
+  Collateral Type: %s
+  Collateral:      %s
+`, p.Title, p.Description, p.CollateralType, p.Collateral))
+	return b.String()
+}
+
+// ValidateBasic stateless validation of the proposal.
+func (p *CommunityCDPWithdrawCollateralProposal) ValidateBasic() error {
```

Comment:
should be able to copy-pasta with minor adjustments the unit tests for validate basic & stringer from other cdp proposal.

per the proverbial merge checklist:
> All public methods have unit tests

https://github.com/Kava-Labs/kava/blob/afc38bb2e5e26ee028bd84d7db1b90f238df7993/x/community/types/proposal_test.go#L138

Reply:
ah yes missed this file, thanks 😄